python/pyspark/sql/dataframe.py

```diff
@@ -1762,7 +1762,7 @@ def toPandas(self):
         else:
```

Comment:
If we use this approach, how about the following to check if the type corrections are needed:

```python
dtype = {}
for field in self.schema:
    pandas_type = _to_corrected_pandas_type(field.dataType)
    if pandas_type is not None and not(field.nullable and pdf[field.name].isnull().any()):
        dtype[field.name] = pandas_type
```